prime-router/src/main/kotlin/fhirengine/utils/FHIRBundleHelpers.kt

```diff
@@ -59,6 +59,28 @@ object FHIRBundleHelpers {
         if (targetList.isNotEmpty()) provenanceResource.target.addAll(targetList)
     }
 
+    internal fun addProvenanceReference(fhirBundle: Bundle) {
+        val provenanceResource = try {
+            fhirBundle.entry.first { it.resource.resourceType.name == "Provenance" }.resource as Provenance
+        } catch (e: NoSuchElementException) {
+            throw IllegalStateException("The FHIR bundle does not contain a Provenance resource")
+        }
```

Comment:
The catch is missing code coverage.

Reply:
Added an additional test for manipulating bundles without provenance.